crates/autopilot/src/domain/settlement/transaction/mod.rs

```diff
@@ -156,6 +190,28 @@ fn is_settlement_trace(trace: &eth::CallFrame, settlement_contract: eth::Address
         && trace.input.0.starts_with(&*SETTLE_FUNCTION_SELECTOR)
 }
 
+async fn find_solver_address<T: Authenticator>(
+    authenticator: &T,
+    path: Vec<&eth::CallFrame>,
+) -> Result<Option<eth::Address>, Error> {
+    let valid_solvers: Vec<Option<eth::Address>> =
+        // The RPC calls to check each address can be done in parallel as they are cheap
+        futures::future::join_all(path.iter().map(|call| async {
```

Comment:
nit: This works just as well with the added benefit of destructing the vec ASAP.
```suggestion
        futures::future::join_all(path.into_iter().map(|call| async {
```